server/src/main/java/org/opensearch/env/EnvironmentSettingsResponse.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.env;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.transport.TransportResponse;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Environment Settings Response for Extensibility
+ *
+ * @opensearch.internal
+ */
+public class EnvironmentSettingsResponse extends TransportResponse {
```

Comment:
Still not clear what the whole EnvironmentSettings request / response is doing: we never use `getEnvironmentSettings` (except in tests)

Reply:
Hi @reta! We are using this response class in the SDK repo [here](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/handlers/EnvironmentSettingsResponseHandler.java#L49).